main.go

```diff
@@ -42,88 +45,408 @@ func main() {
 	pr, _, err := client.PullRequests.Get(ctx, *owner, *repo, *prNumber)
 	if err != nil {
 		fmt.Printf("Error fetching PR details: %v\n", err)
 		os.Exit(1)
 	}
 
-	// Generate a review using GPT-4
-	review, err := generateReviewWithAssistant(pr)
+	// Fetch the current user (the reviewer)
+	user, _, err := client.Users.Get(ctx, "")
+	if err != nil {
+		fmt.Printf("Error fetching user details: %v\n", err)
+		os.Exit(1)
+	}
+
+	// Fetch PR checks (e.g., CI tests)
+	checks, _, err := client.Checks.ListCheckRunsForRef(ctx, *owner, *repo, *pr.Head.SHA, &github.ListCheckRunsOptions{})
+	if err != nil {
+		fmt.Printf("Error fetching PR checks: %v\n", err)
+		os.Exit(1)
+	}
+
+	// If any check has failed, do not allow approval
+	checksPassed := true
+	for _, check := range checks.CheckRuns {
+		if check.GetConclusion() == "failure" {
+			checksPassed = false
+			break
+		}
+	}
+
+	// Fetch PR files
+	files, _, err := client.PullRequests.ListFiles(ctx, *owner, *repo, *prNumber, &github.ListOptions{})
+	if err != nil {
+		fmt.Printf("Error fetching PR files: %v\n", err)
+		os.Exit(1)
+	}
+
+	// Check for pending reviews
+	pendingReview, err := getPendingReview(client, ctx, *owner, *repo, *prNumber)
+	if err != nil {
+		fmt.Printf("Error checking for pending reviews: %v\n", err)
+		os.Exit(1)
+	}
+
+	// Handle existing pending review
+	if pendingReview != nil {
+		fmt.Println("A pending review already exists.")
+		if *dryRun {
+			fmt.Println("Dry run: Review not posted to GitHub.")
+			return
+		}
+
+		// Optionally, submit or dismiss the pending review here
+		// For now, we'll dismiss it to proceed with the new review
+		err = dismissPendingReview(client, ctx, *owner, *repo, *prNumber, pendingReview.GetID(), "Dismissing pending review to submit a new one.")
+		if err != nil {
+			fmt.Printf("Error dismissing pending review: %v\n", err)
+			os.Exit(1)
+		}
+	}
+
+	// Generate the review and parse the assistant's recommendation
+	review, reviewComments, action, err := generateReviewWithAssistant(pr, files)
 	if err != nil {
 		fmt.Printf("Error generating review: %v\n", err)
 		os.Exit(1)
 	}
 
 	// Output the generated review
-	fmt.Println("Generated Review:")
-	fmt.Println(review)
+	log.Println("------- Generated Review:")
+	log.Println(review)
+	log.Println(`------- File comments:`)
+
+	for _, comment := range reviewComments {
+		log.Printf("File: %s, Line: %d\nComment: %s\n", *comment.Path, *comment.Position, *comment.Body)
+	}
+
+	log.Println(`-------`)
 
-	// Post the review if a flag is set
-	if *approve || *requestChanges {
-		err = postReview(client, ctx, *owner, *repo, *prNumber, review, *approve)
+	if *dryRun {
+		// If dry run, don't post the review, just output the results
+		log.Println("Dry run: Review not posted to GitHub.")
+		return
+	}
+
+	// Check if the reviewer is the PR author
+	isSelfReview := user.GetLogin() == pr.User.GetLogin()
+
+	if isSelfReview {
+		// Post the review as a comment instead
+		commentBody := review
```

Comment:
Confirm that user comparisons handle the case where the user may not be logged in properly.